File: security_cam/lib/sensors/audio_classifier_factory.dart
import 'dart:io';

import '../detection/audio/audio_classifier.dart';
import '../detection/audio/yamnet_audio_event_classifier.dart';

/// Audio classifier factory.
///
/// On mobile the real YAMNet model is used ([YamnetAudioEventClassifier]);
/// on desktop the [MockAudioEventClassifier] keeps the pipeline exercisable
/// without bundling native LiteRT + a model into dev-time builds.
Future<AudioEventClassifier> buildAudioClassifier() async {
  if (Platform.isAndroid || Platform.isIOS) {
    return YamnetAudioEventClassifier.load();
  }
  return MockAudioEventClassifier();
}

File: android/app/src/main/kotlin/io/securitycam/level2/detection/audio/AudioClassifierFactory.kt
package io.securitycam.level2.detection.audio

import android.content.Context

/**
 * Audio classifier factory.
 * Uses the real YAMNet model when it loads; degrades to the mock classifier so
 * monitoring keeps working without the model.
 */
object AudioClassifierFactory {
    suspend fun build(context: Context): AudioEventClassifier =
        YamnetClassifier.load(context) ?: MockAudioEventClassifier()
}